File: src/src/hvsun/level_2/뒤에있는큰수찾기.js
function solution(numbers) {
    const copyArr = [...numbers]
    const resultArr = []
    
    let result
    for (let i=0; i<copyArr.length; i++) {
        result = -1
        if (numbers[i-1] < copyArr[i]) result = copyArr[i]
        copyArr.shift()
    }
    resultArr.push(result)
    
    return resultArr
}

File: src/src/bmkim/level_2/뒤에있는큰수찾기.java

public class 뒤에있는큰수찾기 {
    public static void main(String[] args) {
        int[] numbers = {2, 3, 3, 5};
        solution(numbers);
    }

    public static int[] solution(int[] numbers) {
        int[] answer = new int[numbers.length];

        for (int i = 0; i < numbers.length; i++) {
            boolean flag = false;
            for (int j = i + 1; j < numbers.length; j++) {
                if (numbers[j] > numbers[i]) {
                    answer[i] = numbers[j];
                    flag = true;
                    break;
                }
            }

            if (!flag) {
                answer[i] = -1;
            }
        }
        return answer;
    }
}
